master/buildbot/test/util/scheduler.py

```diff
@@ -204,7 +204,7 @@ def fake_addBuildsetForSourceStampsWithDefaults(self, reason, sourcestamps=None,
         self.assertIsInstance(sourcestamps, list)
 
         def sourceStampKey(sourceStamp):
-            return tuple(sorted(sourceStamp.values()))
```

Comment:
values() is not reliable order, so we only order by codebase, which should be unique and defined in the sourcestamps